src/shared/components/case-viewer/case-viewer.component.ts

```diff
@@ -94,8 +96,8 @@ export class CaseViewerComponent implements OnInit, OnDestroy {
   private init() {
     // Clone and sort tabs array
     this.sortedTabs = this.orderService.sort(this.caseDetails.tabs);
-
-    this.caseFields = this.getTabFields();
+    let tabFields = this.getTabFields()
+    this.caseFields = this.reinstantiateCaseFields(tabFields);
```

Comment:
don't think manually reinstantiating the case fields is the right thing to do. We should instead look at the origin of those fields, and why they were not class instances at the origin. And fix it there.
For example could we apply the class transform library to `case-tab.model.ts`? Finding out why that model class gets deserialised and do a `plainToClass` there. In that case, all this code introduced here is no longer needed

Reply:
removed and moved code to calling urls